unstructured_ingest/v2/processes/connectors/confluence.py

```diff
@@ -134,54 +133,201 @@ def precheck(self) -> bool:
+                        # edit permission implies view permission
+                        if action == "update":
+                            doc_perm_entry["roles"].add("read")
+
+            for space_perm in space_permissions:
+                space_operation = space_perm["operation"]["key"]
+                space_target_type = space_perm["operation"]["targetType"]
+                space_entity_id = space_perm["principal"]["id"]
+                space_entity_type = space_perm["principal"]["type"]
+
+                if space_target_type == "space":
+                    if space_operation == "read" and not page_view_restricted:
+                        # Apply space-level view permissions if no page restrictions exist
+                        perm_entry = normalized_perms_dict.setdefault(
+                            space_entity_id,
+                            {
+                                "type": space_entity_type,
+                                "id": space_entity_id,
+                                "email_address": None,
+                                "roles": set(),
+                            },
+                        )
+                        perm_entry["roles"].add("read")
+
+                    elif space_operation == "administer":
+                        # Administer permission includes view + edit
+                        if not page_view_restricted:
+                            perm_entry = normalized_perms_dict.setdefault(
+                                space_entity_id,
+                                {
+                                    "type": space_entity_type,
+                                    "id": space_entity_id,
+                                    "email_address": None,
+                                    "roles": set(),
+                                },
+                            )
+                            perm_entry["roles"].add("read")
+                            if not page_edit_restricted:
+                                perm_entry["roles"].add("update")
+
+                # Add the "delete page" space permissions if there are other page permissions
+                elif (
+                    space_target_type == "page"
+                    and space_operation == "delete"
+                    and space_entity_id in normalized_perms_dict
+                ):
+                    normalized_perms_dict[space_entity_id]["roles"].add("delete")
+
+            # turn sets into sorted lists for consistency and json serialization
+            for perm_data in normalized_perms_dict.values():
+                perm_data["roles"] = sorted(perm_data["roles"])
+
+            return list(normalized_perms_dict.values())
+
+        with self.connection_config.get_client() as client:
+            try:
+                # space permissions
+                space_permissions = []
+                space_permissions_result = client.get(f"/api/v2/spaces/{space_id}/permissions")
```

Comment:
Just asking(maybe not possible), is there any way to get any of this data in the original request to download the doc? (just to reduce the number of api calls)